boot/bootutil/src/loader.c

```diff
@@ -1039,6 +1040,12 @@ boot_go(struct boot_rsp *rsp)
 
     switch (swap_type) {
     case BOOT_SWAP_TYPE_NONE:
+#ifdef BOOTUTIL_VALIDATE_SLOT0
```

Comment:
Then again, if your build system defines these symbols to to the empty string, this won't work.  I suppose `#ifdef` is more standard.